Frameworks/CoreGraphics/CGContextCairo.mm

```diff
@@ -513,14 +513,24 @@
 
     // Give array of advances of zero so it will use positions correctly
     std::vector<FLOAT> dwriteAdvances(count, 0);
-
     DWRITE_GLYPH_RUN run = { fontFace.Get(), curState->fontSize, count, glyphs, dwriteAdvances.data(), positions.data(), FALSE, 0 };
+
+    // CGContextShow* functions transform the glyphs about the origin of the first glyph, rather than about each glyph's origin
+    CGAffineTransform textMatrix{ curState->curTextMatrix };
+    CGAffineTransform ctm{ curState->curTransform };
+    curState->curTransform = CGAffineTransformConcat(textMatrix, ctm);
+    curState->curTextMatrix = CGAffineTransformIdentity;
+
     CGContextDrawGlyphRun(&run);
-    curState->curTextPosition = { curState->curTextPosition.x + delta.x, curState->curTextPosition.y + delta.y };
+
+    // Return transforms to previous values, move text position to end of last glyph drawn
+    curState->curTransform = ctm;
+    curState->curTextMatrix = textMatrix;
```

Comment:
Read further, still not super pleased with this but thinking it might be more elegant than alternatives >:/

---
In reply to: [91807145](https://github.com/Microsoft/WinObjC/pull/1520#discussion_r91807145) [](ancestors = 91807145,91782996)